demos/http/http_demo_plaintext/http_demo_plaintext.c

```diff
@@ -156,6 +203,63 @@ static int sendHttpRequest( const TransportInterface_t * pTransportInterface,
 
 /*-----------------------------------------------------------*/
 
+static int connectToServerWithBackoffRetries( NetworkContext_t * pNetworkContext )
```

Comment:
This code is duplicated in the library - it should be refactored into common logic which can be linked into both examples rather than having 2 almost identical copies.  The only thing that differs are the details of the connect function; the TLS version takes slightly different parameters.  I would create a function type which can be used in either variant, then the individual demos would call a common routine and pass a pointer to the connect function as well as the connect parameters.